File: app/services/api/src/dtos/input/course.py
from typing import Dict, Optional

from pydantic import BaseModel, Field

from app.libs.db import models


class CreateCourse(BaseModel):
    """
    DTO for creating a new course
    """
    institution_id: str
    name: str


class UpdateCourse(BaseModel):
    """
    DTO for updating an existing course
    """
    name: Optional[str] = None

File: app/services/ui/src/types/course.ts
export interface CourseData {
  id?: string;
  _id?: string;
  institution_id: string;
  name: string;
}

export interface CourseShape {
  id: string;
  institution_id: string;
  name: string;
}

export class Course implements CourseShape {
  id: string;
  institution_id: string;
  name: string;

  constructor(data: CourseData) {
    this.id = data.id ?? data._id ?? '';
    this.institution_id = data.institution_id;
    this.name = data.name;
  }

  static from(data: CourseData): Course {
    return new Course(data);
  }
}
